kamu-adapter-http/src/ws_smart_transfer_protocol_tungstenite_client.rs

```diff
@@ -20,14 +20,11 @@ use url::Url;
 
 use tokio_tungstenite::{connect_async, tungstenite::Message, MaybeTlsStream, WebSocketStream};
 
-use kamu::domain::{
-    Dataset, ErrorIntoInternal, GetRefError, InternalError, InvalidIntervalError,
-    ResultIntoInternal, SyncError, SyncListener, SyncResult,
-};
-use kamu::{domain::BlockRef, infra::utils::smart_transfer_protocol::SmartTransferProtocolClient};
+use kamu::domain::*;
+use kamu::infra::utils::smart_transfer_protocol::SmartTransferProtocolClient;
 
 use crate::{
-    dataset_protocol_helper::{dataset_import_object_file, dataset_import_pulled_metadata},
+    dataset_protocol_helper::*,
```

Comment:
As an idea - we could change crate structure to clearly separate by protocol type and client vs server.

E.g. users would do `use kamu_adapter_http::smart_transfer_protocol::client::*` and within submodules we won't have to use verbose names like `ws_smart_transfer_protocol_tungstenite_client` and `SmartProtocolPullClientError` - protocol and client server differences would be in the "namespace" rather than types.

Reply:
Partially reorganized the structure of modules, and closed a lot of redundand public accesses.
The users now have a convenient type alias for the client implementation, and invoke like this:

`kamu_adapter_http::SmartTransferProtocolClientImpl`